src/java/voldemort/client/protocol/admin/AdminClient.java

```diff
@@ -299,6 +301,28 @@ public void close() {
     }
 
     /**
+     * If Cluster is modified ( nodes are added/removed), using the old
+     * AdminClient will cause inconsistent operations. This method returns true
+     * under such cases, so that caller can discard the current AdminClient and
+     * create new AdminClient.
+     * 
+     * @return true if cluster is modified since the AdminClient is create,
+     *         false otherwise
+     */
+    public boolean isClusterModified() {
```

Comment:
I don't think that we should expose leak this complexity to the code which uses the `AdminClient`.

The `ZenStoreClient` refreshes its `cluster.xml` on its own. Why can't the `AdminClient` do the same?